vllm/entrypoints/openai/serving_chat.py

```diff
@@ -208,11 +238,27 @@ async def chat_completion_full_generator(
         assert final_res is not None
 
         choices = []
+
         role = self.get_chat_request_role(request)
         for output in final_res.outputs:
+            token_ids = output.token_ids
+            top_logprobs = output.logprobs
+
+            if request.logprobs is not None:
+                assert(top_logprobs is not None),\
```

Comment:
Should better move this to `protocol.py`. Return 400 when this condition isn't met.